packages/go_router/doc/navigation.md

```diff
@@ -0,0 +1,56 @@
+There are many ways to navigate between destinations in your app.
+
+## Go directly to a destination
+Navigating to a destination in GoRouter will replace the current stack of screens with the screens configured to be displayed
+for the destination route. To change to a new screen, call `context.go()` with a URL:
+
+```
+context.go('/users/123')
+```
```

Comment:
I'm worried that a newbie won't know what `context` is. Should we have an example that includes a widget's `build` method to show where the `context` came from? Or should we call out that this `context` is the `BuildContext` with a link to some `BuildContext doc?

Reply:
Added a realistic build method:

```
build(BuildContext context) {
  return TextButton(
    onPressed: () => context.go('/users/123'),
  );
}
```

BuildContext is not probably something new users understand, but it's something they will encounter right away, so they probably know what it is and what it's used for.